plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidViewHolderRefactoring.java

```diff
@@ -0,0 +1,312 @@
+                        // check whether inflate is inside a conditional
+                        // assignment
+                        || (inflateExpr != null && inflateExpr.getNodeType() == ASTNode.CONDITIONAL_EXPRESSION);
+            }
+            return false;
+        }
+
+        public Expression getInflateExpr() {
+            if (this.viewVariableDeclarationFragment != null) {
+                return this.viewVariableDeclarationFragment.getInitializer();
+            } else if (this.viewVariableAssignment != null) {
+                return this.viewVariableAssignment.getRightHandSide();
+            }
+            return null;
+        }
+
+        private boolean isInflateMethod(MethodInvocation node) {
+            return isMethod(node, "android.view.LayoutInflater", "inflate", "int", "android.view.ViewGroup")
+                    || isMethod(node, "android.view.LayoutInflater", "inflate", "int", "android.view.ViewGroup",
+                            "boolean")
+                    || isMethod(node, "android.view.LayoutInflater", "inflate", "org.xmlpull.v1.XmlPullParser",
+                            "android.view.ViewGroup")
+                    || isMethod(node, "android.view.LayoutInflater", "inflate", "org.xmlpull.v1.XmlPullParser",
+                            "android.view.ViewGroup", "boolean");
+        }
+    }
+
+    static class FindViewByIdVisitor extends ASTVisitor {
+        private List<FindViewByIdItem> items = new ArrayList<FindViewByIdItem>();
+
+        static class FindViewByIdItem {
+            private SimpleName variable;
+            private Expression findViewByIdExpr;
+            private VariableDeclarationFragment findViewByIdDeclarationFragment;
+            private Assignment findViewByIdVariableAssignment;
+            private MethodInvocation findViewByIdInvocation;
+
+            FindViewByIdItem(MethodInvocation node) {
+                setAssignment(node);
+            }
+
+            private void setAssignment(MethodInvocation node) {
+                this.findViewByIdInvocation = node;
+                ASTNode ancestor = getFirstAncestorOrNull(node, VariableDeclarationFragment.class, Assignment.class);
+                if (ancestor instanceof VariableDeclarationFragment) {
+                    this.findViewByIdDeclarationFragment = (VariableDeclarationFragment) ancestor;
+                    this.variable = this.findViewByIdDeclarationFragment.getName();
+                    this.findViewByIdExpr = this.findViewByIdDeclarationFragment.getInitializer();
+                } else if (ancestor instanceof Assignment) {
+                    this.findViewByIdVariableAssignment = (Assignment) ancestor;
+                    ASTNode leftSide = this.findViewByIdVariableAssignment.getLeftHandSide();
+                    if (leftSide.getNodeType() == SIMPLE_NAME) {
+                        this.variable = (SimpleName) leftSide;
```

Comment:
So now instead of a `ClassCastException`, we risk a `NullPointerException` at line 164: `item.variable.getIdentifier()));`